### PR TITLE
Ruleset implementation

### PR DESCRIPTION
**Pull request by lado936**
### Description of your changes

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: internal/controller/repository/repository.go [hunk 1]
```diff
@@ -198,6 +199,22 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
...
+		ghRepositoryRules, _ := getRepositoryRules(ctx, c.github, cr.Spec.ForProvider.Org, name)
+
+		crRepositoryRulesToConfig := getRepositoryRulesMapFromCr(cr.Spec.ForProvider.RepositoryRules)
+		ghRepositoryRulesToConfig, err := getRepositoryRulesWithConfig(ctx, c.github, cr.Spec.ForProvider.Org, name, ghRepositoryRules)
+		if err != nil {
+			return managed.ExternalObservation{}, err
+		}
+
+		if !cmp.Equal(crRepositoryRulesToConfig, ghRepositoryRulesToConfig) {
+			fmt.Println(cmp.Diff(crRepositoryRulesToConfig, ghRepositoryRulesToConfig))
```
Comment (tkaesserfm) [inline internal/controller/repository/repository.go:212]: Prints can be removes

File: internal/controller/repository/repository.go [hunk 2]
```diff
@@ -977,6 +1006,448 @@ func handleBranchProtectionSignature(ctx context.Context, gh *ghclient.Client, o
...
+				}
+			}
+			if rRules.PullRequest != nil {
+				rRules.PullRequest.DismissStaleReviewsOnPush = util.BoolDerefToPointer(rRules.PullRequest.DismissStaleReviewsOnPush, false)
+				rRules.PullRequest.RequireCodeOwnerReview = util.BoolDerefToPointer(rRules.PullRequest.RequireCodeOwnerReview, false)
+				rRules.PullRequest.RequireLastPushApproval = util.BoolDerefToPointer(rRules.PullRequest.RequireLastPushApproval, false)
+				rRules.PullRequest.RequiredReviewThreadResolution = util.BoolDerefToPointer(rRules.PullRequest.RequiredReviewThreadResolution, false)
+				rRules.PullRequest.RequiredApprovingReviewCount = util.IntDerefToPointer(rRules.PullRequest.RequiredApprovingReviewCount, 0)
+			}
+			//if rRules.RequiredStatusChecks == nil {
```
Comment (tkaesserfm) [inline internal/controller/repository/repository.go:1108]: What's up with this commented out code?

File: internal/controller/repository/repository.go [hunk 3]
```diff
@@ -977,6 +1006,448 @@ func handleBranchProtectionSignature(ctx context.Context, gh *ghclient.Client, o
...
+						params := github.RequiredDeploymentEnvironmentsRuleParameters{}
+						if err := json.Unmarshal(*rule.Parameters, &params); err != nil {
+							return nil, err
+						}
+						ruleset.Rules.RequiredDeployments = &v1alpha1.RulesRequiredDeployments{
+							Environments: util.SortAndReturn(params.RequiredDeploymentEnvironments),
+						}
+					}
+				case "required_status_checks":
+					fmt.Println("required_status_checksAAAAAAAAAAAAA")
```
Comment (tkaesserfm) [inline internal/controller/repository/repository.go:1230]: please remove :)

**Review by tkaesserfm — CHANGES_REQUESTED** [review]

File: internal/controller/repository/repository.go (hunk 3 above)
Comment (tkaesserfm) [reply internal/controller/repository/repository.go:1230]: the print i mean :)

File: internal/controller/repository/repository.go (hunk 2 above)
Comment (lado936) [reply internal/controller/repository/repository.go]: i was testing status checks functionality but that commented stuff is not needed anymore, forgot to remove

**Review by tkaesserfm — APPROVED** [review]
LGTM! Thank you so much for your contribution @lado936 !

**Pull request merged**